$OpenBSD: patch-vendor_bundle_jruby_1_9_gems_filewatch-0_6_1_lib_filewatch_watch_rb,v 1.1 2015/04/26 19:02:57 jasper Exp $

Unbreak due to NotImplementedError: stat.st_dev unsupported or native support failed to load
https://logstash.jira.com/browse/LOGSTASH-665

--- vendor/bundle/jruby/1.9/gems/filewatch-0.6.1/lib/filewatch/watch.rb.orig	Mon Mar 30 20:50:11 2015
+++ vendor/bundle/jruby/1.9/gems/filewatch-0.6.1/lib/filewatch/watch.rb	Mon Mar 30 20:50:51 2015
@@ -76,7 +76,7 @@ module FileWatch
           fileId = Winhelper.GetWindowsUniqueFileIdentifier(path)
           inode = [fileId, stat.dev_major, stat.dev_minor]
         else
-          inode = [stat.ino.to_s, stat.dev_major, stat.dev_minor]
+          inode = [stat.ino.to_s, nil, nil]
         end
 		
         if inode != @files[path][:inode]
@@ -149,7 +149,7 @@ module FileWatch
         stat = File::Stat.new(file)
         @files[file] = {
           :size => 0,
-          :inode => [stat.ino, stat.dev_major, stat.dev_minor],
+          :inode => [stat.ino, nil, nil],
           :create_sent => false,
         }
 		
@@ -157,7 +157,7 @@ module FileWatch
           fileId = Winhelper.GetWindowsUniqueFileIdentifier(file)
           @files[file][:inode] = [fileId, stat.dev_major, stat.dev_minor]
         else
-          @files[file][:inode] = [stat.ino.to_s, stat.dev_major, stat.dev_minor]
+          @files[file][:inode] = [stat.ino.to_s, nil, nil]
         end
 		
         if initial
